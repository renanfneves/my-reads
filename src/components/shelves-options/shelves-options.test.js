import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ShelvesOptions from './shelves-options';

const defaultProps = {
  id:"mockedId",
  changeShelf:() => {}
};

it('renders shelf with default value', () => {
  const renderer = new ShallowRenderer();

  renderer.render(
    <ShelvesOptions
      {...defaultProps}
    />,
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});

it('renders shelf with read value', () => {
  const renderer = new ShallowRenderer();

  renderer.render(
    <ShelvesOptions
      {...defaultProps}
      currentShelf="read"
    />,
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});

it('renders shelf with currentlyReading value', () => {
  const renderer = new ShallowRenderer();

  renderer.render(
    <ShelvesOptions
      {...defaultProps}
      currentShelf="currentlyReading"
    />,
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});

it('renders shelf with wantToRead value', () => {
  const renderer = new ShallowRenderer();

  renderer.render(
    <ShelvesOptions
      {...defaultProps}
      currentShelf="wantToRead"
    />,
  );

  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
