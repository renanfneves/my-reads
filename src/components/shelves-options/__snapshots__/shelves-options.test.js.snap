// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders shelf with currentlyReading value 1`] = `
<select
  onChange={[Function]}
  value="currentlyReading"
>
  <option
    disabled={true}
    value=""
  >
    Move to...
  </option>
  <option
    value="currentlyReading"
  >
    Currently Reading
  </option>
  <option
    value="wantToRead"
  >
    Want to Read
  </option>
  <option
    value="read"
  >
    Read
  </option>
  <option
    value="none"
  >
    None
  </option>
</select>
`;

exports[`renders shelf with default value 1`] = `
<select
  onChange={[Function]}
  value="none"
>
  <option
    disabled={true}
    value=""
  >
    Move to...
  </option>
  <option
    value="currentlyReading"
  >
    Currently Reading
  </option>
  <option
    value="wantToRead"
  >
    Want to Read
  </option>
  <option
    value="read"
  >
    Read
  </option>
  <option
    value="none"
  >
    None
  </option>
</select>
`;

exports[`renders shelf with read value 1`] = `
<select
  onChange={[Function]}
  value="read"
>
  <option
    disabled={true}
    value=""
  >
    Move to...
  </option>
  <option
    value="currentlyReading"
  >
    Currently Reading
  </option>
  <option
    value="wantToRead"
  >
    Want to Read
  </option>
  <option
    value="read"
  >
    Read
  </option>
  <option
    value="none"
  >
    None
  </option>
</select>
`;

exports[`renders shelf with wantToRead value 1`] = `
<select
  onChange={[Function]}
  value="wantToRead"
>
  <option
    disabled={true}
    value=""
  >
    Move to...
  </option>
  <option
    value="currentlyReading"
  >
    Currently Reading
  </option>
  <option
    value="wantToRead"
  >
    Want to Read
  </option>
  <option
    value="read"
  >
    Read
  </option>
  <option
    value="none"
  >
    None
  </option>
</select>
`;
